crates/sys/platform/src/linux_arm.rs

```diff
@@ -1,59 +1,49 @@
 //! Obtain particular CPU features for AArch64 on Linux
 
+use core::sync::atomic::{AtomicBool, Ordering};
 use libc::{getauxval, AT_HWCAP, HWCAP_AES, HWCAP_ASIMD, HWCAP_PMULL, HWCAP_SHA2};
 
 #[inline(always)]
 fn auxval() {
     let val = unsafe { getauxval(AT_HWCAP) };
-    let val_asimd = val & HWCAP_ASIMD != 0;
-    unsafe { ADV_SIMD = val_asimd };
     let val_aes = val & HWCAP_AES != 0;
-    unsafe { AES = val_aes };
+    AES.store(val_aes, Ordering::Relaxed);
     let val_pmull = val & HWCAP_PMULL != 0;
-    unsafe { PMULL = val_pmull };
+    PMULL.store(val_pmull, Ordering::Relaxed);
     let val_sha256 = val & HWCAP_SHA2 != 0;
-    unsafe { SHA256 = val_sha256 };
+    SHA256.store(val_sha256, Ordering::Relaxed);
 }
 
-static mut ADV_SIMD: bool = false;
-static mut AES: bool = false;
-static mut PMULL: bool = false;
-static mut SHA256: bool = false;
+static AES: AtomicBool = AtomicBool::new(false);
+static PMULL: AtomicBool = AtomicBool::new(false);
+static SHA256: AtomicBool = AtomicBool::new(false);
 
 #[inline(always)]
 pub(super) fn aes() -> bool {
     init();
-    unsafe { AES }
-}
-
-#[inline(always)]
-pub(super) fn adv_simd() -> bool {
-    init();
-    unsafe { ADV_SIMD }
+    AES.load(Ordering::Relaxed);
```

Comment:
```suggestion
    AES.load(Ordering::Relaxed)
```

Thank you! It looks like there's a superfluous `;` here.